api/v1alpha1/gateway_types.go

```diff
@@ -180,6 +188,66 @@ type ListenerTLS struct {
 	Options map[string]string `json:"options"`
 }
 
+// TLSTerminationPolicy defines the schema of a TLS termination policy.
+type TLSTerminationPolicy struct {
+	// TerminationType defines how to terminate TLS connections.
+	//
+	// If unspecified, TLS termination type "Edge" will be used.
+	//
+	// Support: Core
+	//
+	// +optional
+	TerminationType TLSTerminationType `json:"terminationType,omitempty"`
+	// Routes is one or more routes to associate with the TLS termination policy.
+	//
+	// If unspecified, all routes associated with the Gateway listener will be used.
+	//
+	// Support: Core
+	//
+	// +optional
+	Routes []core.ObjectReference `json:"routes,omitempty"`
+	// CACertificates is a reference to one or more CA certificate used
+	// for establishing a TLS connection with the final destination when
+	// using TLS termination type "Reencrypt".
+	//
+	// Here is a ConfigMap example (in yaml):
+	//
+	// apiVersion: v1
+	// kind: ConfigMap
+	// metadata:
+	//  name: my-dest-svc-ca
+	//  namespace: my-dest-svc-namespace
+	//  data:
+	//    ca-bundle.crt: |
+	//      -----BEGIN CERTIFICATE-----
+	//      Destination Service CA Certificate Bundle.
+	//      -----END CERTIFICATE-----
+	//
+	// Support: Core (Kubernetes ConfigMap)
+	// Support: Implementation-specific (For other resource types)
+	//
+	// +optional
+	CACertificates []core.TypedLocalObjectReference `json:"caCertificates,omitempty"`
```

Comment:
Should we require specific secret types for this? (IMHO, yes we should)

Reply:
Since this is server TLS policy for the listener, I think that we should not also be specifying the client TLS policy for re-ecrypting traffic to the backend. That client policy could have different validation policy, version requirements, client certificates, and all sorts of things that are awkward to express here.

Reply:
@jpeach I'm pushing a new commit where the TLS termination policy is associated to a `Route` within a `Gateway`.